src/main/java/com/influxdb/v3/client/internal/VectorSchemaRootConverter.java

```diff
@@ -46,94 +47,151 @@
-                    default:
-                    case NANOSECOND:
-                        timeUnit = TimeUnit.NANOSECONDS;
-                        break;
-                }
-                long nanoseconds = TimeUnit.NANOSECONDS.convert((Long) value, timeUnit);
-                pointValues.setTimestamp(Instant.ofEpochSecond(0, nanoseconds));
+    /**
+     * Function to cast value return base on metadata from InfluxDB.
+     *
+     * @param field the Field object from Arrow
+     * @param value the value to cast
+     * @return the casted value
+     */
+    public Object getMappedValue(@Nonnull final Field field, @Nullable final Object value) {
+        if (value == null) {
+            return null;
+        }
+
+        var fieldName = field.getName();
+        if ("measurement".equals(fieldName) || "iox::measurement".equals(fieldName)) {
+            return value.toString();
+        }
+
+        var metaType = field.getMetadata().get("iox::column::type");
+        if (metaType == null) {
+            if ("time".equals(fieldName) && (value instanceof Long || value instanceof LocalDateTime)) {
+                return NanosecondConverter.getTimestampNano(value, field);
             } else {
-                pointValues.setTimestamp(Instant.ofEpochMilli((Long) value));
+                return value;
             }
-        } else if (value instanceof LocalDateTime) {
-            pointValues.setTimestamp(((LocalDateTime) value).toInstant(ZoneOffset.UTC));
         }
+
+        String[] parts = metaType.split("::");
+        String valueType = parts[2];
+        if ("field".equals(valueType)) {
+            switch (metaType) {
+                case "iox::column_type::field::integer":
+                case "iox::column_type::field::uinteger":
+                    if (value instanceof Number) {
+                        return TypeCasting.toLongValue(value);
+                    } else {
+                        LOG.warning(String.format("Value of %s is not an Integer", fieldName));
+                        return value;
+                    }
+                case "iox::column_type::field::float":
+                    if (value instanceof Number) {
+                        return TypeCasting.toDoubleValue(value);
+                    } else {
+                        LOG.warning(String.format("Value of %s is not a Double", fieldName));
+                        return value;
+                    }
+                case "iox::column_type::field::string":
+                    if (value instanceof Text || value instanceof String) {
+                        return TypeCasting.toStringValue(value);
+                    } else {
+                        LOG.warning(String.format("Value of %s is not a String", fieldName));
+                        return value;
+                    }
+                case "iox::column_type::field::boolean":
+                    if (value instanceof Boolean) {
+                        return (Boolean) value;
```

Comment:
The casting here is not necessary:

```suggestion
                        return value;
```